### PR TITLE
Exception messages

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java [hunk 1]
```diff
@@ -415,6 +415,25 @@ public QueryExecutionAssert failsWithMessage(String expectedErrorMessage)
 
             return this;
         }
+
+        public QueryExecutionAssert failsWithMessageMathing(String expectedErrorMessagePattern)
```
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java]: typo s/Mathing/Matching

File: tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java [hunk 2]
```diff
@@ -415,6 +415,25 @@ public QueryExecutionAssert failsWithMessage(String expectedErrorMessage)
...
+            String exceptionMessage = executionException.getMessage();
+            LOGGER.debug("Query failed as expected, with message: {}", exceptionMessage);
+            if (!nullToEmpty(exceptionMessage).matches(expectedErrorMessagePattern)) {
+                throw new AssertionError(format(
+                        "Query failed with unexpected error message: '%s' \n Expected error message to match '%s'",
+                        exceptionMessage,
+                        expectedErrorMessagePattern));
+            }
+
+            return this;
```
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java]: this method is copy-paste of method above, maybe you could extract common code

File: tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java (hunk 1 above)
Comment (findepi) [reply tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java]: good catch!

File: tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java (hunk 2 above)
Comment (findepi) [reply tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java]: good idea, done

**Pull request merged**